### PR TITLE
feat(get-app-authentication): improve error message when privateKey is incomplete

### LINKED ISSUE #311: Improve error message when `privateKey` is not a single line String
**Context:** [40:09 in Reusing GitHub Apps and Actions with Gregor Martynus](https://www.twitch.tv/videos/1102860651?t=40m09s)

---

I can take it once we agree on how to tackle this :)

### PR DESCRIPTION
**Pull request by oscard0m**
## 📝 Summary

- To improve error message when `privateKey` provided by a user is incomplete (with only `-----BEGIN RSA PRIVATE KEY-----`)

## ⛱ Motivation and Context

Fix #311

### CODE REVIEW

**Comment by oscard0m** [comment]
@gr2m Not sure if we should test this in a pre-release first. What do you think?

File: test/index.test.ts [hunk 1]
```diff
@@ -93,6 +93,36 @@ test("throws if invalid 'type' is provided", async () => {
...
+
+test("throws if incomplete Private Key is provided", async () => {
+  const auth = createAppAuth({
+    appId: APP_ID,
+    privateKey: "-----BEGIN RSA PRIVATE KEY-----",
+  });
+
+  await expect(auth({ type: "app" })).rejects.toEqual(
+    new Error(
+      "Private key is incomplete. Make sure it is a single line String and newlines have been replaced by '\n'"
```
Comment (gr2m) [inline test/index.test.ts:121]: Looks good! I suggest we slightly change the error message to

```suggestion
      "The 'privateKey` option contains only the first line '-----BEGIN RSA PRIVATE KEY-----'. If you are setting it using a `.env` file, make sure it is set on a single line with newlines replaced by '\n'"
```

what do you think?

**Comment by gr2m** [comment]
> @gr2m Not sure if we should test this in a pre-release first. What do you think?

no pre-release necessary

can you check what's up with the node 10 tests?

**Comment by oscard0m** [comment]
> > @gr2m Not sure if we should test this in a pre-release first. What do you think?
> 
> no pre-release necessary
> 
> can you check what's up with the node 10 tests?

checking Node 10 issue

**Review by gr2m — APPROVED** [review]

**Comment by gr2m** [comment]
Thanks Oscar!

**Pull request merged**